plugins/module_utils/network/sonic/config/vrrp/vrrp.py

```diff
@@ -0,0 +1,758 @@
+        'virtual_address': 'vrrp/vrrp-group={vrid}/config/virtual-address',
+        'track_interface': 'vrrp/vrrp-group={vrid}/openconfig-interfaces-ext:vrrp-track'
+    }
+
+    vrrp_attributes = ('preempt', 'version', 'use_v2_checksum', 'priority', 'advertisement_interval', 'virtual_address', 'track_interface')
+
+    def __init__(self, module):
+        super(Vrrp, self).__init__(module)
+
+    def get_vrrp_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        vrrp_facts = facts['ansible_network_resources'].get('vrrp')
+        if not vrrp_facts:
+            return []
+        return vrrp_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = list()
+        existing_vrrp_facts = self.get_vrrp_facts()
+        commands, requests = self.set_config(existing_vrrp_facts)
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+
+        result['before'] = existing_vrrp_facts
+        old_config = existing_vrrp_facts
+
+        if self._module.check_mode:
+            result.pop('after', None)
+            new_config = self._get_generated_config(commands, existing_vrrp_facts, self._module.params['state'])
+            self.sort_lists_in_config(new_config)
+            result['after(generated)'] = new_config
+        else:
+            changed_vrrp_facts = self.get_vrrp_facts()
+            new_config = changed_vrrp_facts
+            self.sort_lists_in_config(new_config)
+            if result['changed']:
+                result['after'] = changed_vrrp_facts
+
+        if self._module._diff:
+            self.sort_lists_in_config(old_config)
+            self.sort_lists_in_config(new_config)
+            result['diff'] = get_formatted_config_diff(old_config, new_config, self._module._verbosity)
+
+        result['commands'] = commands
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_vrrp_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
```

Comment:
Please normalize interfaces names specified in 'name' and 'track_interfaces' options.